### PR TITLE
Added Binary Insertion Sort in Java

### LINKED ISSUE #2125: Binary Insertion Sort
- [x] C
- [x] C++
- [x] C# - @Yashank18
- [ ] CoffeeScript
- [ ] Dart - @raksha009
- [ ] Go - @KamalDGRT
- [ ] Java - @raktim00
- [ ] JS - @sheetalssr
- [ ] Kotlin
- [ ] Php - @goyalmuskan
- [x] Python
- [ ] Readme - @goyalmuskan
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by raktim00**
Fixes #2125 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Binary Insertion Sort in Java

#### Languages Used:
- Java

#### Files Added:
- BinaryInsertionSort.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Binary_Insertion_Sort/BinaryInsertionSort.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (jainaman224) [inline Binary_Insertion_Sort/BinaryInsertionSort.java]: Fix file name.

File: Binary_Insertion_Sort/BinaryInsertionSort.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * Binary Insertion Sort
```
Comment (jainaman224) [inline Binary_Insertion_Sort/BinaryInsertionSort.java]: Remove this.

**Comment by somya-kapoor** [comment]
@raktim00 Are you working on it?

**Comment by raktim00** [comment]
Changes done.

File: Binary_Insertion_Sort/Binary_Insertion_Sort.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Author: Raktim Midya 
+ * Date: 05/03/2020
+*/
```
Comment (AdityaShD) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.java]: remove these lines

File: Binary_Insertion_Sort/Binary_Insertion_Sort.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        s.close();
+
+        new BinaryInsertionSort().bis(arr);
+
+        System.out.print("The Sorted Array is: ");
+        for (int i = 0; i < arr.length; i++)
+            System.out.print(arr[i] + " ");
+    }
+
+    public void bis(int array[]) {
```
Comment (AdityaShD) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.java]: fix function name, use something else instead of bis

**Pull request closed without merging**